lib/rack/mock.rb

```diff
@@ -257,9 +257,13 @@ def identify_cookie_attributes(cookie_filling)
       cookie_bits.drop(1).each do |bit|
         if bit.include? '='
           cookie_attribute, attribute_value = bit.split('=', 2)
-          cookie_attributes.store(cookie_attribute.strip, attribute_value.strip)
           if cookie_attribute.include? 'max-age'
             cookie_attributes.store('expires', Time.now + attribute_value.strip.to_i)
+          elsif cookie_attribute.include? 'expires'
+            next if cookie_attributes.key? 'expires' # max-age attribute has precedence
+            cookie_attributes.store('expires', Time.httpdate(attribute_value.strip))
+          else
+            cookie_attributes.store(cookie_attribute.strip, attribute_value.strip)
           end
```

Comment:
I wonder if a better algorithm for this would be to avoid checking for `max-age` and `expires` inside the `cookie_bits.drop(1).each` loop.  Instead, after the loop, if `cookie_attributes['max-age']` is set, then set `cookie_attributes['expires']`.  If it isn't set and `cookie_attributes['expires']` is set, convert it to a `Time`.

Reply:
Yes, that will definitely be easier to understand.

Reply:
Pushed new version, I kept the "style" of the old code for the sake of consistency. E.g. using `#store` over `#[]=`.